apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -128,6 +148,31 @@ export class SubpoenaService {
       )
     }
 
+    if (
```

Comment:
You could simplify this code by creating a const thats either a NotificationType or undefined depending on if/which of these 3 cases are true. Then use that to decide whether to send a message of said type to the queue. That way you dont need to perform the same checks twice and you know both whether to send and which type to send with the same check.